### PR TITLE
Adds support for storage_pairs_paged

### PR DESCRIPTION
**Pull request by crystalin**
Adds the RPC methods to query the storage pairs (key, value) with pagination.
The goal is to provide a faster way to iterate through keys and their values. Previous method was to query the keys first and their values in additional queries.

### CODE REVIEW

File: client/rpc/src/state/mod.rs [hunk 1]
```diff
@@ -239,13 +248,31 @@ where
...
 	}
 
+	fn storage_pairs_paged(
+		&self,
+		prefix: Option<StorageKey>,
+		count: u32,
+		start_key: Option<StorageKey>,
+		block: Option<Block::Hash>,
+	) -> RpcResult<Vec<(StorageKey, StorageData)>> {
+		if count > STORAGE_KEYS_PAGED_MAX_COUNT {
```
Comment (librelois) [inline client/rpc/src/state/mod.rs:265]:
```suggestion
        self.deny_unsafe.check_if_safe()?;
		if count > STORAGE_KEYS_PAGED_MAX_COUNT {
```

**Comment by crystalin** [comment]
@librelois Now I think about it, I'm not sure it is should be considered "unsafe", the goal is accelerate the "get_keys_paged + query_storage_at" which are both considered "safe".

**Comment by librelois** [comment]
> @librelois Now I think about it, I'm not sure it is should be considered "unsafe", the goal is accelerate the "get_keys_paged + query_storage_at" which are both considered "safe".

Yes, I suggested to make it unsafe because `state_getPairs` is unsafe, but maybe it is reasonable to consider this new method safe because it is paginated.

**Review by librelois — APPROVED** [review]

**Comment by bkchr** [comment]
We are working (slowly) on adding new and better RPC interface: https://github.com/paritytech/json-rpc-interface-spec/

So, I would restrain from adding new RPC functionality. You can keep it in your own fork for now. I hope this is fine.

**Comment by crystalin** [comment]
snif :(
How far are we from this new interface ?

**Comment by bkchr** [comment]
There are already some implementations in https://github.com/paritytech/substrate/tree/master/client/rpc-spec-v2

CC @lexnv

**Pull request closed without merging**